src/txn/txn.c

```diff
@@ -717,14 +717,12 @@ __wt_txn_release(WT_SESSION_IMPL *session)
 }
 
 /*
- * __txn_locate_hs_record --
- *     Locate the update older than the prepared update in the history store and append it to the
- *     update chain if necessary.
+ * __txn_restore_hs_update --
+ *     Restore the history store update to the update chain
  */
 static int
-__txn_locate_hs_record(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, WT_PAGE *page,
-  WT_UPDATE *chain, bool commit, WT_UPDATE **fix_updp, bool *upd_appended,
-  WT_UPDATE *first_committed_upd, bool first_committed_upd_in_hs)
+__txn_restore_hs_update(
```

Comment:
This function now only does one thing to restore the history store update to the update chain if we have a prepared rollback.

Reply:
The update we allocate here in this function should be always append to the update chain. Therefore, there is no need to free the update anymore. As in the previous implementation, we may allocate a temporary update but not append it to pass on the timestamp information to the later code.